### ISSUE TITLE
render bug in infinite-scroll-js

### ISSUE THREAD

**Issue by gyrevik** [KunalN25/my-tutorials#1]
I forked the repo and ran infinite-scroll-js locally. I started scrolling through it and a few seconds into that it started blinking and jerking around.

**Comment by KunalN25** [comment]
> I forked the repo and ran infinite-scroll-js locally. I started scrolling through it and a few seconds into that it started blinking and jerking around.

Hi @gyrevik, thanks for pointing this out. The issue came up because after a certain no of pages (around 10), the API returned an empty array, and my code wasn't handling that case. Since the content was still intersecting with the viewport, it kept making API calls.

I have added a `hasMore` state to prevent this. It should work properly now.

**Comment by gyrevik** [comment]
I don't see the update yet for some reason.

**Comment by KunalN25** [comment]
Yeah, I had checked again, my first attempt didn't fix the issue, so I hid the comment. But, the issue is fixed now

**Comment by gyrevik** [comment]
Thanks for fixing it man!
